dlt/destinations/impl/duckdb/duck.py

```diff
@@ -87,20 +87,28 @@ def to_db_integer_type(
         )
 
     def to_db_datetime_type(
-        self, precision: Optional[int], table_format: TTableFormat = None
+        self,
+        timezone: Optional[bool],
+        precision: Optional[int],
+        table_format: TTableFormat = None,
     ) -> str:
+        # TIMESTAMP and TIMESTAMPTZ(TIMESTAMP WITH TIME ZONE) supports microsecond precision
+        if timezone:
```

Comment:
here I wound not ignore wrong combinations of precision and timezone for duckdb.

if we see that user wants timezone but precision does not support it we should raise exception where we explain exactly what is going on (must include table name, column name) and how to fix it (change the timezone to False)